File: A2Z/go/sorting/1/bubble_sort_test.go
package sortone_test

import (
	"reflect"
	"sortone"
	"testing"
)

func TestBubbleSort(t *testing.T) {
	tests := []struct {
		input []int
		want  []int
	}{
		{
			input: []int{4, 1, 3, 9, 7},
			want:  []int{1, 3, 4, 7, 9},
		},
		{
			input: []int{10, 9, 8, 7, 6, 5, 4, 3, 2, 1},
			want:  []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
		},
		{
			input: []int{1, 2, 3, 4, 5},
			want:  []int{1, 2, 3, 4, 5},
		},
	}

	for _, test := range tests {
		got := sortone.BubbleSort(test.input)
		if !reflect.DeepEqual(test.want, got) {
			t.Errorf("want %v, got %v", test.want, got)
		}
	}
}

File: A2Z/js/sorting/1/bubble-sort.test.mjs
import { expect, test } from "vitest";
import Solution from "./bubble-sort.mjs";

test("test bubble sort", () => {
  const tests = [
    {
      input: [4, 1, 3, 9, 7],
      want: [1, 3, 4, 7, 9],
    },
    {
      input: [10, 9, 8, 7, 6, 5, 4, 3, 2, 1],
      want: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    },
    {
      input: [1, 2, 3, 4, 5],
      want: [1, 2, 3, 4, 5],
    },
  ];

  const solution = new Solution();

  for (const test of tests) {
    const got = solution.bubbleSort(test.input);
    expect(got).toEqual(test.want);
  }
});
